File: python/bogosort.py
import random
import sys
import time

def Sorted(arr):
	for i in range(len(arr)-1):
		if arr[i] > arr[i+1]:
			return False
	return True
	
def Shuffle(arr):
	for i in range(len(arr)):
		otherindex = random.randint(0, len(arr)-1)		
		holder = arr[otherindex]
		arr[otherindex] = arr[i]
		arr[i] = holder

def Bogosort(arr):
	while not Sorted(arr):
		Shuffle(arr)
		
def RandArray(length = 10, maxvalue=255):
	arr = []
	for i in range(length):
		arr.append(random.randint(0, maxvalue))
	return arr
	
if __name__ == '__main__':
	length = 10
	maxvalue = 255
	count = 1
	verbose = False
	arr = []
	individualtimes = []
	
	letter = ""
	i = 1
	while i < len(sys.argv):
		'''Waiting for python3.10
		match sys.argv[i]:
			case "-l":
				length = int(sys.argv[i+1])
			case "-m":
				maxvalue = int(sys.argv[i+1])
			case "-c":
				count = int(sys.argv[i+1])
		i+=1
		'''
	
		if sys.argv[i] == "-l":
			length = int(sys.argv[i+1])
			i+=1
		if sys.argv[i] == "-m":
			maxvalue = int(sys.argv[i+1])
			i+=1
		if sys.argv[i] == "-c":
			count = int(sys.argv[i+1])
			i+=1
		if sys.argv[i] == "-v":
			verbose = True
		if sys.argv[i] == "-a":
			while i+1 < len(sys.argv) and sys.argv[i+1].isnumeric():
				arr.append(int(sys.argv[i+1]))
				i+=1
		i+=1
		
	if arr != []:
		starttime = time.time()
		Bogosort(arr)
		totaltime = time.time() - starttime
		for number in arr:
			print(str(number) + ", ", end="")
	else:
		starttime = time.time()
		for i in range(count):
			if verbose:
				individualstarttime = time.time()
			arr = RandArray(length, maxvalue)
			Bogosort(arr)
			if verbose:
				individualtimes.append(time.time() - individualstarttime)
				print("time " + str(i) + ":", float(int(((time.time() - individualstarttime))*1000))/1000)
		totaltime = time.time() - starttime
		if verbose:
			summeduptotaltime = 0
			for time in individualtimes:
				summeduptotaltime += time
				print("time:", float(int(time*1000))/1000)
			print("average time:", float(int((summeduptotaltime/count)*1000))/1000)
			print("summed up time:", float(int(summeduptotaltime*1000))/1000)
		print("total time:", float(int(totaltime*1000))/1000)

File: javascript/bogosort.js
function Bogosort(Arr)
{
	while(!IsSorted(Arr))
	{
		Shuffle(Arr);
	}
	return Arr;
};

function Shuffle(Arr){
	for(let i = 0; i < Arr.length; i++)
	{
		let OtherIndex = Math.floor(Math.random() * Arr.length);
		let Holder = Arr[i];
		Arr[i] = Arr[OtherIndex];
		Arr[OtherIndex] = Holder;
	}
	return Arr;
};

function IsSorted(Arr){
	for(let i = 0; i < Arr.length - 1; i++)
	{
		if(Arr[i] > Arr[i+1])
		{
			return false;
		}
	}
	return true;
};

function RandomArray(Len = 5, Max = 255){
	let Arr = [];
	for(let i = 0; i < Len; i++)
	{
		Arr.push(Math.floor(Math.random() * Max + 1));
	}
	return Arr;
};

let Verbose = false;
let Len = 8;
let Max = 255;
let Count = 1;
let Arr = [];

for(i = 1; i < process.argv.length; i++)
{
	switch(process.argv[i])
	{
		case "-v":
			Verbose = true;
			break;
		case "-l":
			i++;
			Len = Number(process.argv[i]);
			break; 
		case "-m":
			i++;
			Max = Number(process.argv[i]);
			break; 
		case "-c":
			i++;
			Count = Number(process.argv[i]);
			break;
		case "-a":
			let Var = "0";
			while(!isNaN(Var) && i < process.argv.length - 1)
			{
				Var = process.argv[i+1].replace(",", "");
				Arr.push(Number(Var));
				i++;
			}
			break;
	}
}

if(Arr.length > 0)
{
	Arr = Bogosort(Arr);
	let First = true;
	let Result = "";
	Arr.forEach(element => {
		if(!First)
		{
			Result += ", ";
		}
		else
		{
			First = false;
		}
		Result += element;
	});
	console.log(Result);
}
else
{
	let IndividualTimes = [];

	let Start = new Date().getTime();

	let IndividualStart = 0;

	for(i = 0; i < Count; i++)
	{
		if(Verbose)
		{
			IndividualStart = new Date().getTime();
		}
		Bogosort(RandomArray(Len, Max));
		if(Verbose)
		{
			IndividualTimes.push((new Date().getTime() - IndividualStart) / 1000);
			console.log( i + ": " + IndividualTimes[i] + "s");
		}
	}

	if(Verbose)
	{
		let SummedUpTime = 0;
		for(i = 0; i < IndividualTimes.length; i++)
		{
			SummedUpTime += IndividualTimes[i];
		}
		console.log("Average time: " + (SummedUpTime / IndividualTimes.length) + "s");
		console.log("Summed up time: " + SummedUpTime + "s");
	}

	let End = new Date().getTime();
	console.log("Total: " + ((End - Start)/1000) + "s");
}